### PR TITLE
fix: remove disabled flags from bulk evaluation

### LINKED ISSUE #652: [BUG] Bulk API evaluates disabled flags
### Observed behavior

Calls to the bulk evaluation API include disabled flags.

The following error is also written to the console:

```
playground-flagd-1           | 2023-05-11T17:53:38.965Z debug   eval/json_evaluator.go:263      requested flag is disabled: new-color-scheme    {"requestID": "chein8gulbm2ltkfoi80", "component": "evaluator", "evaluator": "json"}
playground-flagd-1           | 2023-05-11T17:53:38.965Z error   eval/json_evaluator.go:142      bulk evaluation: key: new-color-scheme returned error: FLAG_DISABLED    {"requestID": "chein8gulbm2ltkfoi80", "component": "evaluator", "evaluator": "json"}
playground-flagd-1           | github.com/open-feature/flagd/core/pkg/eval.(*JSONEvaluator).ResolveAllValues
playground-flagd-1           |  /workspace/core/pkg/eval/json_evaluator.go:142
playground-flagd-1           | github.com/open-feature/flagd/core/pkg/service/flag-evaluation.(*FlagEvaluationService).ResolveAll
playground-flagd-1           |  /workspace/core/pkg/service/flag-evaluation/flag_evaluator.go:60
playground-flagd-1           | github.com/bufbuild/connect-go.NewUnaryHandler[...].func1
playground-flagd-1           |  /go/pkg/mod/github.com/bufbuild/connect-go@v1.7.0/handler.go:52
playground-flagd-1           | github.com/bufbuild/connect-opentelemetry-go.(*Interceptor).WrapUnary.func1
playground-flagd-1           |  /go/pkg/mod/github.com/bufbuild/connect-opentelemetry-go@v0.1.0/interceptor.go:145
playground-flagd-1           | github.com/bufbuild/connect-go.NewUnaryHandler[...].func2
playground-flagd-1           |  /go/pkg/mod/github.com/bufbuild/connect-go@v1.7.0/handler.go:76
playground-flagd-1           | github.com/bufbuild/connect-go.(*Handler).ServeHTTP
playground-flagd-1           |  /go/pkg/mod/github.com/bufbuild/connect-go@v1.7.0/handler.go:238
playground-flagd-1           | net/http.(*ServeMux).ServeHTTP
playground-flagd-1           |  /usr/local/go/src/net/http/server.go:2500
playground-flagd-1           | github.com/open-feature/flagd/core/pkg/service/middleware/metrics.Middleware.Handler.func1.1
playground-flagd-1           |  /workspace/core/pkg/service/middleware/metrics/http_metrics.go:105
playground-flagd-1           | github.com/open-feature/flagd/core/pkg/service/middleware/metrics.Middleware.Measure
playground-flagd-1           |  /workspace/core/pkg/service/middleware/metrics/http_metrics.go:90
playground-flagd-1           | github.com/open-feature/flagd/core/pkg/service/middleware/metrics.Middleware.Handler.func1
playground-flagd-1           |  /workspace/core/pkg/service/middleware/metrics/http_metrics.go:104
playground-flagd-1           | net/http.HandlerFunc.ServeHTTP
playground-flagd-1           |  /usr/local/go/src/net/http/server.go:2122
playground-flagd-1           | github.com/rs/cors.(*Cors).Handler.func1
playground-flagd-1           |  /go/pkg/mod/github.com/rs/cors@v1.9.0/cors.go:236
playground-flagd-1           | net/http.HandlerFunc.ServeHTTP
playground-flagd-1           |  /usr/local/go/src/net/http/server.go:2122
playground-flagd-1           | golang.org/x/net/http2/h2c.h2cHandler.ServeHTTP
playground-flagd-1           |  /go/pkg/mod/golang.org/x/net@v0.9.0/http2/h2c/h2c.go:125
playground-flagd-1           | net/http.serverHandler.ServeHTTP
playground-flagd-1           |  /usr/local/go/src/net/http/server.go:2936
playground-flagd-1           | net/http.(*conn).serve
playground-flagd-1           |  /usr/local/go/src/net/http/server.go:1995
```

### Expected Behavior

Disabled flags should be excluded from bulk evaluation.

### Steps to reproduce

Start flagd with the following flags:

```json
{
	"flags": {
		"new-welcome-message": {
			"state": "ENABLED",
			"defaultVariant": "on",
			"variants": {
				"on": true,
				"off": false
			}
		},
		"new-color-scheme": {
			"state": "DISABLED",
			"defaultVariant": "off",
			"variants": {
				"on": true,
				"off": false
			}
		}
	}
}
```

Now cURL the resolve all endpoint http://localhost:8013/schema.v1.Service/ResolveAll

### PR DESCRIPTION
**Pull request by Kavindu-Dodan**
## This PR

Fixes #652 & enhance tests to validate error sates 

Note - had to sync go deps and thus the go mod/sum changes

### CODE REVIEW

File: core/pkg/service/flag-evaluation/flag_evaluator_test.go [hunk 1]
```diff
@@ -903,3 +903,50 @@ func getMetricReader() (*telemetry.MetricsRecorder, metric.Reader) {
 	rs := resource.NewWithAttributes("testSchema")
 	return telemetry.NewOTelRecorder(exp, rs, "testSvc"), exp
 }
+
+// TestFlag_Evaluation_ErrorCodes test validate error mapping from known errors to connect.
+// Code and avoid accidental changes. This is essential as SDK implementations rely on connect.
+// Code to differentiate GRPC errors vs Flag errors. For any change, we must change respective SDK changes
```
Comment (toddbaert) [inline core/pkg/service/flag-evaluation/flag_evaluator_test.go:909]:
```suggestion
// TestFlag_Evaluation_ErrorCodes test validate error mapping from known errors to connect.
// Code and avoid accidental changes. This is essential as provider implementations rely on connect.
// Code to differentiate gRPC errors vs Flag errors. Changes must be absorbed by flagd providers.
```

Comment (toddbaert) [reply core/pkg/service/flag-evaluation/flag_evaluator_test.go:909]: The capitalization here is a bit odd. Maybe a block comment would be better?

File: core/pkg/eval/json_evaluator.go [hunk 2]
```diff
@@ -122,6 +122,11 @@ func (je *JSONEvaluator) ResolveAllValues(ctx context.Context, reqID string, con
 	var err error
 	allFlags := je.store.GetAll()
 	for flagKey, flag := range allFlags {
+		if flag.State == Disabled {
+			// ignore evaluation of disabled flag
+			continue
+		}
+
```
Comment (toddbaert) [inline core/pkg/eval/json_evaluator.go:129]: Is there any testing for this? It's probably worth adding.

File: core/pkg/service/flag-evaluation/flag_evaluator_test.go [hunk 3]
```diff
@@ -903,3 +903,50 @@ func getMetricReader() (*telemetry.MetricsRecorder, metric.Reader) {
...
+func TestFlag_Evaluation_ErrorCodes(t *testing.T) {
```
Comment (toddbaert) [inline core/pkg/service/flag-evaluation/flag_evaluator_test.go:910]: These are useful tests to add, good idea.

**Comment by Kavindu-Dodan** [comment]
@toddbaert  sorry, I forgot to switch to main before branching for this fix, hence it contains changes inteded for #671 

I will reopen a PR with clean change set

**Pull request closed without merging**